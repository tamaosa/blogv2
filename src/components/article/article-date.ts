const format = (d: Date) => {
  return `${d.getFullYear()}年${d.getMonth() + 1}月${d.getDate()}日`
}

export const articleDate = (published: string, updated?: string) => {
  const p = format(new Date(published))
  const u = updated ? format(new Date(updated)) : undefined
  return u ? `${p} 公開 / ${u} 更新` : `${p} 公開`
}