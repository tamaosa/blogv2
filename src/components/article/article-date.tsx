import React from "react"
import { css } from "@emotion/react"

import { rhythm } from "../../utils/typography"

const format = (d: Date) => {
  return `${d.getFullYear()}年${d.getMonth() + 1}月${d.getDate()}日`
}

const dateStyel = css`
  display: inline-block;
  margin-right: ${rhythm(1 / 4)};
`

type Props = {
  published: string
  updated?: string
}

export const ArticleDate: React.FC<Props> = ({ published, updated }) => {
  const p = format(new Date(published))
  const u = updated ? format(new Date(updated)) : undefined
  return (
    <>
      <span css={dateStyel}>
        <time>{p}</time> 公開
      </span>
      {u && (
        <span css={dateStyel}>
          {" "}
          <time>{u}</time> 更新
        </span>
      )}
    </>
  )
}
