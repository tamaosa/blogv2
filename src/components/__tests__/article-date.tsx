import React from "react"
import { render } from "@testing-library/react"
import { ArticleDate } from "../article-date"

describe("ArticleDate", () => {
  it("snapshot test1", () => {
    const { asFragment } = render(<ArticleDate published="2022-01-01" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("snapshot test2", () => {
    const { asFragment } = render(
      <ArticleDate published="2022-01-01" updated="2022-12-31" />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("matching publication date", () => {
    const { container } = render(<ArticleDate published="2022-01-01" />)
    expect(container).toHaveTextContent("2022年1月1日 公開")
  })

  it("matching update date", () => {
    const { container } = render(
      <ArticleDate published="2022-01-01" updated="2022-12-31" />
    )
    expect(container).toHaveTextContent("2022年12月31日 更新")
  })
})
