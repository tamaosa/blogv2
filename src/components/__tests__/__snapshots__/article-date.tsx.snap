// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDate snapshot test1 1`] = `
<DocumentFragment>
  <span
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
  >
    <time>
      2022年1月1日
    </time>
     公開
  </span>
</DocumentFragment>
`;

exports[`ArticleDate snapshot test2 1`] = `
<DocumentFragment>
  <span
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
  >
    <time>
      2022年1月1日
    </time>
     公開
  </span>
  <span
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
  >
     
    <time>
      2022年12月31日
    </time>
     更新
  </span>
</DocumentFragment>
`;
